src/e2e/java/teammates/e2e/cases/AdminNotificationsPageE2ETest.java

```diff
@@ -0,0 +1,75 @@
+import teammates.common.util.AppUrl;
+import teammates.common.util.Const;
+import teammates.common.util.TimeHelper;
+import teammates.e2e.pageobjects.AdminNotificationsPage;
+
+/**
+ * SUT: {@link Const.WebPageURIs#ADMIN_NOTIFICATIONS_PAGE}.
+ */
+public class AdminNotificationsPageE2ETest extends BaseE2ETestCase {
+    private NotificationAttributes[] notifications = new NotificationAttributes[2];
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/AdminNotificationsPageE2ETest.json");
+        removeAndRestoreDataBundle(testData);
+
+        notifications[0] = testData.notifications.get("notification1");
+        notifications[1] = testData.notifications.get("notification2");
+    }
+
+    @Test
+    @Override
+    public void testAll() {
+
+        ______TS("verify loaded data");
+        AppUrl url = createFrontendUrl(Const.WebPageURIs.ADMIN_NOTIFICATIONS_PAGE);
+        AdminNotificationsPage notificationsPage = loginAdminToPage(url, AdminNotificationsPage.class);
+        notificationsPage.verifyNotificationsTable(notifications);
+        verifyPresentInDatabase(notifications[0]);
+        verifyPresentInDatabase(notifications[1]);
+
+        ______TS("add new notification");
+        NotificationAttributes newNotification = NotificationAttributes
+                .builder("tm.e2e.ANotifications.notif1")
+                .withStartTime(TimeHelper.parseInstant("2035-04-01T22:00:00Z"))
+                .withEndTime(TimeHelper.parseInstant("2035-04-30T20:00:00Z"))
+                .withStyle(NotificationStyle.SUCCESS)
+                .withTargetUser(NotificationTargetUser.GENERAL)
+                .withTitle("E2E test notification 1")
+                .withMessage("<p>E2E test notification message</p>")
+                .build();
+
+        notificationsPage.addNotification(newNotification);
+        notificationsPage.verifyStatusMessage("Notification created successfully.");
+        String newestNotificationId = notificationsPage.getNewestNotificationId();
+        NotificationAttributes createdNotification = getNotification(newestNotificationId);
+        notificationsPage.verifyNotificationAttributes(newNotification, createdNotification);
+
+        ______TS("edit notification");
+        createdNotification.setStartTime(TimeHelper.parseInstant("2025-05-09T12:00:00Z"));
+        createdNotification.setEndTime(TimeHelper.parseInstant("2045-06-01T22:00:00Z"));
+        createdNotification.setStyle(NotificationStyle.DANGER);
+        createdNotification.setTargetUser(NotificationTargetUser.INSTRUCTOR);
+        createdNotification.setTitle("Edited E2E test notification 1");
+        createdNotification.setMessage("<p>Edited E2E test notification message</p>");
+
+        notificationsPage.editNotification(createdNotification);
+        notificationsPage.verifyStatusMessage("Notification updated successfully.");
+        verifyPresentInDatabase(createdNotification);
+
```

Comment:
Verifying that the edited notification is present in DB is not sufficient. Need to check that the notification details are correctly updated (both in the front end and in the backend). The frontend verification should be straightforward, but the backend verification might need a workaround since we don't have `verifyUpdatedInDatabase`, the trick is to reload the page using `reloadPage()` from `AppPage.java` and then verify the details on the frontend again.

Reply:
Thank you for the suggestions! I have made improvements on the testcases, and in the process also found and fixed a bug where calling `verifyPresentInDatabase` causes the `createdAt` and `updatedAt` attributes of the input notification to become null.